onebusaway-api-webapp/src/main/java/org/onebusaway/api/actions/api/gtfs_realtime/AlertsForAgencyAction.java

```diff
@@ -61,5 +74,18 @@ protected void fillFeedMessage(FeedMessage.Builder feed, String agencyId,
       ServiceAlertBuilderHelper.fillFeedMessage(feed, alerts, agencyId, timestamp);
     }
 
+    if(feedTimestamp != 0){
+      long lastModifiedMills = feedTimestamp * 1000L;
+      String lastModifiedHeader = _sdf.format(new Date(lastModifiedMills));
+      HttpServletResponse response = ServletActionContext.getResponse();
+      response.setHeader("Last-Modified", lastModifiedHeader);
+    } else {
+      long lastModifiedMills = timestamp * 1000L;
+      String lastModifiedHeader = _sdf.format(new Date(lastModifiedMills));
+
+      HttpServletResponse response = ServletActionContext.getResponse();
+      response.setHeader("Last-Modified", lastModifiedHeader);
+    }
+
```

Comment:
Try to have less repetitive code, look into the "DRY" principals (Don't Repeat Yourself):
long lastModifiedMills;
String lastModifiedHeader;

if(feedTimestamp != 0){
      lastModifiedMills = feedTimestamp * 1000L;
    } else {
      lastModifiedMills = timestamp * 1000L;
    }
lastModifiedHeader = _sdf.format(new Date(lastModifiedMills));
HttpServletResponse response = ServletActionContext.getResponse();
response.setHeader("Last-Modified", lastModifiedHeader);

Reply:
Understand, will do. corrected as mentioned.